pkg/builder/local_build_executor.go

```diff
@@ -248,6 +253,13 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 		return response
 	}
 
+	if be.devices != nil {
```

Comment:
I would suggest that `CreateDev()` is simply inlined inside this function. Even though it mutates the file system, it's not generic enough that it warrants having it as a utility function in `re_filesystem`.